### PR TITLE
Added troubleshooting for microbit v2 if openOCD could not find MEM-AP

### PR DESCRIPTION
**Pull request by irina-b-dev**
### Pull Request Overview

This pull request adds troubleshooting section to the microbit v2 readme. Some of the microbits have a debugging protection active that prevents openOCD from controlling the chip.

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: boards/microbit_v2/README.md [hunk 1]
```diff
@@ -127,3 +127,31 @@ $ tockloader --openocd --board microbit_v2 --bundle-apps install app.tab
...
+
+```bash
+$ openocd -f openocd.cfg -c "dap apreg 1 0x04 0x01"
+```
+
+Reset the device. 
+
+> Note that this will erase all Flash and RAM as this is the only way to disable CTRL-AP protection.
+
+> Another solution would be to use Nordic's [nrfjprog](https://infocenter.nordicsemi.com/topic/com.nordic.infocenter.tools/dita/tools/nrf5x_command_line_tools/nrf5x_command_line_tools_lpage.html) to unlock your chip via `$ nrfjprog -f NRF52 --recover`, but this requires the use of a JLink debugger (on-board nRF5 DKs or external).
```
Comment (brghena) [inline boards/microbit_v2/README.md:157]:
```suggestion
> Another solution would be to use Nordic's [nrfjprog](https://infocenter.nordicsemi.com/topic/com.nordic.infocenter.tools/dita/tools/nrf5x_command_line_tools/nrf5x_command_line_tools_lpage.html) to unlock your chip via `$ nrfjprog -f NRF52 --recover`, but this requires the use of a JLink debugger (on-board nRF52 DKs or external).
```

Comment (brghena) [reply boards/microbit_v2/README.md:157]: The nrfjprog link is also bad. Maybe you meant this: https://infocenter.nordicsemi.com/index.jsp?topic=%2Fug_nrf_cltools%2FUG%2Fcltools%2Fnrf_nrfjprogexe.html

File: boards/microbit_v2/README.md [hunk 2]
```diff
@@ -127,3 +127,31 @@ $ tockloader --openocd --board microbit_v2 --bundle-apps install app.tab
 ```
 
 > `--bundle-apps` seems to be needed due to an [openocd issue](https://github.com/tock/tockloader/issues/67)
+
+## Troubleshooting
+
+### Could not find MEM-AP to control the core
+
+OpenOCD displays `Could not find MEM-AP to control the core` error when trying to connect to Micro:Bit. This means that the Micro:boit has [control access port](https://infocenter.nordicsemi.com/topic/com.nordic.infocenter.nrf52832.ps.v1.1/dif.html?cp=2_2_0_15_1#concept_udr_mns_1s) protection enabled.
```
Comment (brghena) [inline boards/microbit_v2/README.md:135]:
```suggestion
OpenOCD displays `Could not find MEM-AP to control the core` error when trying to connect to Micro:bit. This means that the Micro:bit has [control access port](https://infocenter.nordicsemi.com/topic/com.nordic.infocenter.nrf52832.ps.v1.1/dif.html?cp=2_2_0_15_1#concept_udr_mns_1s) protection enabled. 
```

**Review by brghena — CHANGES_REQUESTED** [review]
Thanks for this! Always good to have more troubleshooting information. A few small fixes, but otherwise looks good.

**Comment by brghena** [comment]
I'm surprised that some Micro:bits are coming with debugging protection active. Do you have any insight into how common this is or under what situations someone might run into it?

**Comment by irina-b-dev** [comment]
@brghena Thank you for your feedback! I don't know how common it is for Micro:bits to have debugging protection active.
I did try to use openOCD with two other Micro:bits and they didn't have it enabled.
I might have enabled it by mistake when I used a nightly version of openOCD (I encountered this error afterwards, when using the stable version of openOCD v0.10.0).

File: boards/microbit_v2/README.md [hunk 3]
```diff
@@ -127,3 +127,31 @@ $ tockloader --openocd --board microbit_v2 --bundle-apps install app.tab
...
+
+```bash
+$ openocd -f openocd.cfg -c "dap apreg 1 0x04 0x01"
+```
+
+Reset the device. 
+
+> Note that this will erase all Flash and RAM as this is the only way to disable CTRL-AP protection.
+
+> Another solution would be to use Nordic's [nrfjprog](https://infocenter.nordicsemi.com/topic/com.nordic.infocenter.tools/dita/tools/nrf5x_command_line_tools/nrf5x_command_line_tools_lpage.html) to unlock your chip via `$ nrfjprog -f NRF52 --recover`, but this requires the use of a JLink debugger (on-board nRF52 DKs or external).
```
Comment (brghena) [inline boards/microbit_v2/README.md:157]: The link here is still broken. https://infocenter.nordicsemi.com/topic/com.nordic.infocenter.tools/dita/tools/nrf5x_command_line_tools/nrf5x_command_line_tools_lpage.html resolves into a "Page Not Found" for me.

**Review by brghena — CHANGES_REQUESTED** [review]

File: boards/microbit_v2/README.md (hunk 3 above)
Comment (bradjc) [inline boards/microbit_v2/README.md:157]:
```suggestion
> Another solution would be to use Nordic's [nrfjprog](https://www.nordicsemi.com/Products/Development-tools/nRF-Command-Line-Tools) to unlock your chip via `$ nrfjprog -f NRF52 --recover`, but this requires the use of a JLink debugger (on-board nRF52 DKs or external).
```

**Pull request merged**